src/style-spec/expression/definitions/within.js

```diff
@@ -32,7 +40,7 @@ function boxWithinBox(bbox1: BBox, bbox2: BBox) {
 }
 
 function getTileCoordinates(p, canonical: CanonicalTileID) {
-    const coord = MercatorCoordinate.fromLngLat({lng: p[0], lat: p[1]}, 0);
+    const coord = {x: mercatorXfromLng(p[0]), y: mercatorYfromLat(p[1])};
```

Comment:
Nit: if we're not using `MercatorCoordinate.fromLngLat`, there's no need to wrap the vlaues in an {x, y} object since it's immediately discarded.